independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/parameters/LocatableResourcePathParamExtractor.java

```diff
@@ -0,0 +1,60 @@
+package org.jboss.resteasy.reactive.server.core.parameters;
+
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
+import org.jboss.resteasy.reactive.server.mapping.URITemplate;
+
+public class LocatableResourcePathParamExtractor implements ParameterExtractor {
+
+    private final String name;
+
+    public LocatableResourcePathParamExtractor(String name) {
+        this.name = name;
+    }
+
+    @Override
+    public Object extractParameter(ResteasyReactiveRequestContext context) {
+        int index = findPathParamIndex(context.getLocatorTarget().getClassPath(), context.getLocatorTarget().getPath());
```

Comment:
This appears to imply that you can _only_ access path params in a sub-resource by matching against the locator method, so the sub-resource can't have path params of its own, right?

So, is it true? The spec is confusing about this. If it is true, why do we not keep the original path param values then, instead of adding a new field to save the original ones?

Also, why can't we use `context.getLocatorTarget().getPathParameterIndexes()`? I'm sure there's a good reason, but I don't get it.

Reply:
By reading the spec, I can't really tell what the intended behavior is.

I'll try and see if we can keep the paths of the original request and see how the tests behave

Reply:
I tried keeping the original values and some of the tests fail.

The approach in this PR however passes all old and new tests

Reply:
Then perfect :)